File: 289-game-of-life/289-game-of-life.cpp
class Solution {
public:
    int neborsum(int m, int n, vector<vector<int>>& board, int i, int j){
        int res =0;
        if(i+1!=m)
            res+=board[i+1][j];
        if(i-1!=-1)
            res+=board[i-1][j];
        if(j+1!=n)
            res+=board[i][j+1];
        if(j-1!=-1)
            res+=board[i][j-1];
        
        if(j+1!=n && i+1!=m)
            res+=board[i+1][j+1];
            
        if(j+1!=n && i-1!=-1)
            res+=board[i-1][j+1];
        
        if(j-1!=-1 && i+1!=m)
            res+=board[i+1][j-1];
            
        if(j-1!=-1 && i-1!=-1)
            res+=board[i-1][j-1];
        return res;
    }

    void gameOfLife(vector<vector<int>>& board) {
        vector<vector<int>> finalb = board;
        int m = board.size(), n=board[0].size();
        for(int i=0; i<m; i++){
            for(int j=0; j<n; j++){
                if(board[i][j]==0){
                    int res = neborsum(m,n, board, i, j);
                    if(res==3)
                        finalb[i][j]=1;
                }
                else{
                    int res = neborsum(m,n, board, i, j);
                    if(res<2)
                        finalb[i][j]=0;
                    else if(res>3)
                        finalb[i][j]=0;
                }
            }
        }           
        board = finalb;
    }
};

File: 289-game-of-life/289-game-of-life.cs
public class Solution {
    public void GameOfLife(int[][] board) 
    {
        int n = board.Length, m = board[0].Length;

        for(int i=0; i< n; i++)
        {
            for(int j=0; j< m; j++)
            {
                int neighbour = CountLiveNbrs(board, i, j, n, m);
                if(board[i][j] == 1 && (neighbour < 2 || neighbour > 3))
                {
                    board[i][j] = 2; 
                }
                else if(board[i][j] == 0 && neighbour == 3)
                {
                    board[i][j] = -1; 
                }
            }
        }
        
        for(int i=0; i< n; i++)
        {
            for(int j=0; j< m; j++)
            {
                if(board[i][j] == -1)
                    board[i][j] = 1;
                else if(board[i][j] == 2)
                    board[i][j] = 0;
                 
            }
        }
    }
    
    int CountLiveNbrs(int[][] board, int i, int j, int n, int m)
    {
        int neighbour = 0;
        for(int x = Math.Max(0, i-1); x<Math.Min(n, i+2); x++)
        {
            for(int y = Math.Max(0, j-1); y<Math.Min(m, j+2); y++)
            {
                if(board[x][y] >= 1)
                    neighbour++;
            }
        }
        neighbour -= board[i][j];
        return neighbour;
    }
}